client-js/app/elements/labrad-registry.ts

```diff
@@ -104,7 +129,7 @@ export class LabradRegistry extends polymer.Base {
 
     this.$.pendingDialog.close()
     this.$.combinedList.selected = null;
-
+    this.kick++;
```

Comment:
Computed properties do observe the properties they depend on; I think the underlying problem is a bug in how polymer handles observing array-valued properties.